pkg/platform/baremetal/baremetal.go

```diff
@@ -83,6 +87,10 @@ func NewConfig() *config {
 	}
 }
 
+func (c *config) clusterDomain() string {
```

Comment:
I don't think this function is needed, as it adds extra indirection without any specific reason. For other providers, it is valuable, because it de-duplicates formatting logic.